lib/transforms/loadAssets.js

```diff
@@ -27,22 +27,19 @@ module.exports = function () { // ...
             .parMap(function (assetConfig) {
                 var resolvedAssetConfig = assetGraph.resolveAssetConfig(assetConfig, assetGraph.root);
                 resolvedAssetConfig.isInitial = true;
-                assetGraph.ensureAssetConfigHasType(resolvedAssetConfig, this);
+                assetGraph.ensureAssetConfigHasType(resolvedAssetConfig).nodeify(this);
             })
             .parMap(10, function (resolvedAssetConfig) {
                 var callback = this,
                     asset = assetGraph.createAsset(resolvedAssetConfig);
                 if (asset.keepUnloaded) {
                     callback(null, asset);
                 } else {
-                    asset.load(function (err) {
-                        if (err) {
-                            assetGraph.emit('warn', err);
-                            callback();
-                        } else {
-                            callback(null, asset);
-                        }
-                    });
+                    asset.load().then(function () {
```

Comment:
If `asset.load` resolves with the asset itself this could be reduced to

```js
asset.load().catch(assetGraph.emit.bind(null, 'warn'));
```

Not entirely sure where to plug in `.nodeify(callback)` in that construction. Just at the end?